libs/services/src/events/hatom.enter.market.events.service.ts

```diff
@@ -0,0 +1,122 @@
+import { Injectable } from "@nestjs/common";
+import { EventLog } from "apps/api/src/endpoints/events/entities";
+import { Address } from "@multiversx/sdk-core";
+import BigNumber from "bignumber.js";
+import { CsvRecordsService } from "../records";
+import moment from "moment";
+import { DataService } from "../data";
+import { TableSchema } from "apps/dune-simulator/src/endpoints/dune-simulator/entities";
+import { joinCsvAttributes } from "libs/services/utils";
+
+interface BorrowEvent {
+    eventName: string;
+    moneyMarket: string;
+    borrowerAddress: string;
+    amount: BigNumber;
+}
+
+@Injectable()
+export class HatomEnterMarketEventsService {
+    private readonly headers: TableSchema[] = [
+        { name: 'timestamp', type: 'varchar' },
+        { name: 'money_market', type: 'varchar' },
+        { name: 'token_amount', type: 'double' },
+        { name: 'token_id', type: 'varchar' },
+        { name: 'borrower_address', type: 'varchar' },
+        { name: 'value_in_egld', type: 'double' },
+        { name: 'value_in_usd', type: 'double' },
+    ];
+    constructor(
+        private readonly csvRecordsService: CsvRecordsService,
+        private readonly dataService: DataService,
+    ) { }
+
+    public async hatomEnterMarketWebhook(eventsLog: EventLog[]): Promise<void> {
+
+        for (const eventLog of eventsLog) {
+            const enterMarketEventInHex = '656e7465725f6d61726b65745f6576656e74'; // 'enter_market_event'
+
+            if (eventLog.identifier === "enterMarkets" && eventLog.topics[0] === enterMarketEventInHex) {
```

Comment:
check length of topics before getting topics[0]. we should expect exactly 4 topics in this case